File: AIUniversalAssistantFlutter/lib/utils/extensions.dart
import 'package:intl/intl.dart';

extension DateTimeExtension on DateTime {
  String toDateString() {
    return DateFormat('yyyy-MM-dd HH:mm').format(this);
  }

  String toRelativeDateString() {
    final now = DateTime.now();
    final today = DateTime(now.year, now.month, now.day);
    final yesterday = today.subtract(const Duration(days: 1));
    final dateToCheck = DateTime(year, month, day);

    if (dateToCheck == today) {
      return '今天';
    } else if (dateToCheck == yesterday) {
      return '昨天';
    } else {
      return DateFormat('yyyy-MM-dd').format(this);
    }
  }
}

extension IntExtension on int {
  String formatWords() {
    if (this >= 10000) {
      return '${(this / 10000).toStringAsFixed(0)}万';
    } else if (this >= 1000) {
      return '${(this / 1000).toStringAsFixed(0)}千';
    } else {
      return toString();
    }
  }

  String formatFileSize() {
    if (this <= 0) return '0 KB';
    if (this < 1024) return '$this B';
    if (this < 1024 * 1024) {
      return '${(this / 1024).toStringAsFixed(1)} KB';
    }
    return '${(this / (1024 * 1024)).toStringAsFixed(1)} MB';
  }
}

extension StringExtension on String {
  int countWords() {
    return length;
  }
}

File: AIUniversalAssistantAndroid/app/src/main/java/com/aiwriting/assistant/utils/Extensions.kt
package com.aiwriting.assistant.utils

import android.content.Context
import android.view.View
import android.view.inputmethod.InputMethodManager
import android.widget.Toast
import java.text.SimpleDateFormat
import java.util.*

// Toast 扩展
fun Context.toast(message: String, duration: Int = Toast.LENGTH_SHORT) {
    Toast.makeText(this, message, duration).show()
}

// 显示/隐藏软键盘
fun View.showKeyboard() {
    requestFocus()
    val imm = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
    imm.showSoftInput(this, InputMethodManager.SHOW_IMPLICIT)
}

fun View.hideKeyboard() {
    val imm = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
    imm.hideSoftInputFromWindow(windowToken, 0)
}

// 日期格式化
fun Long.toDateString(pattern: String = "yyyy-MM-dd HH:mm"): String {
    val sdf = SimpleDateFormat(pattern, Locale.getDefault())
    return sdf.format(Date(this))
}

fun Long.toRelativeDateString(): String {
    val now = System.currentTimeMillis()
    val diff = now - this
    
    val calendar = Calendar.getInstance()
    calendar.timeInMillis = this
    val today = Calendar.getInstance()
    
    return when {
        calendar.get(Calendar.YEAR) == today.get(Calendar.YEAR) &&
        calendar.get(Calendar.DAY_OF_YEAR) == today.get(Calendar.DAY_OF_YEAR) -> "今天"
        
        calendar.get(Calendar.YEAR) == today.get(Calendar.YEAR) &&
        calendar.get(Calendar.DAY_OF_YEAR) == today.get(Calendar.DAY_OF_YEAR) - 1 -> "昨天"
        
        else -> this.toDateString("yyyy-MM-dd")
    }
}

// 字数格式化
fun Int.formatWords(): String {
    return when {
        this >= 10000 -> "${this / 10000}万"
        this >= 1000 -> "${this / 1000}千"
        else -> this.toString()
    }
}

// 文件大小格式化
fun Long.formatFileSize(): String {
    return when {
        this <= 0 -> "0 KB"
        this < 1024 -> "$this B"
        this < 1024 * 1024 -> String.format("%.1f KB", this / 1024.0)
        else -> String.format("%.1f MB", this / (1024.0 * 1024.0))
    }
}

// 字数统计
fun String.countWords(): Int {
    return this.length
}

// View 可见性扩展
fun View.visible() {
    visibility = View.VISIBLE
}

fun View.invisible() {
    visibility = View.INVISIBLE
}

fun View.gone() {
    visibility = View.GONE
}

fun View.isVisible(): Boolean = visibility == View.VISIBLE

